src/ucp/core/ucp_ep.h

```diff
@@ -510,8 +517,9 @@ typedef struct ucp_conn_request {
     uct_device_addr_t           *remote_dev_addr;
     struct sockaddr_storage     client_address;
     ucp_ep_h                    ep; /* valid only if request is handled internally */
+    ucp_wireup_user_data_t      user_data;
     ucp_wireup_sockaddr_data_t  sa_data;
-    /* packed worker address follows */
+    ucp_unpacked_address_t      remote_addr;
```

Comment:
can we unpack address where it's needed to locally allocated buffer? I guess `ucp_ep_create_server_accept`

Reply:
It's needed to not do twice address unpack. 
```
    /* [ sa_data | ucp_address | user_data ]
     * |      conn_priv_data_length        |
     * 1. sa_data is fixed of size, so copy it directly */
     * 2. We need to know ucp_address length - so we unpack it not conn_request. Otherwise we would need to traverse packed address and not unpack - just to return it's length. Would complicate a lot
     ```